content/learning/operator/operator.md

```diff
@@ -69,6 +69,54 @@ The kubebuilder framework uses the controller-runtime library to communicate wit
 4. the [request](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Request "request") received in the reconciler function
 5. k8s [corev1](https://godoc.org/k8s.io/api/core/v1 "corev1") in case you need to interact with core k8s kind (pods, deployments, configmaps etc)
 
+## Useful tips
+
+Here are some useful suggestion if you are new to programming an operator:
+1. You may need to watch a second resource. In this case what you need to do is to add Owns in the [setup](https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#setup) function :
+```
+func (r *_ObjectToReconcile_) SetupWithManager(mgr ctrl.Manager) error {
+	
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&principal_resource).
+		Owns(&second resource_to_watch).
+		Complete(r)
+}
+```
+
+2. It is possible to filter the case of reconcile that our operator perform thanks to the [Predicates](https://sdk.operatorframework.io/docs/building-operators/golang/references/event-filtering/). To use them you can modify the setup function explayned above:
+```
+func (r *ObjectToReconcile) SetupWithManager(mgr ctrl.Manager) error {
+	klog.Info("setup manager")
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&principal_resource, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&second_resource_to_watch, builder.WithPredicates(predicate_function)).
+		Complete(r)
+}
+
+func someExamplePredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.UpdateEvent) bool {
+      		// Here you can compare old version and new version of the watched object
+			return condition_to_evaluate
+		},
+      	UpdateFunc: func(e event.UpdateEvent) bool {
+      		// Here you can compare old version and new version of the watched object
+			return condition_to_evaluate
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+      		// Here you can compare old version and new version of the watched object
+			return condition_to_evaluate
+		},
+	}
+}
+```
+
+3. Another problem that you can face is about the  [CreateOrUpdate](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.0/pkg/controller/controllerutil#CreateOrUpdate) function in which you have to create a initial resource (composed only by ObjectMeta's data) before the function's call and, inside it, with different functions, you call a mutation functions in which it is possible to modify the resource fields; this function, is intrinsically idempotent.
+
+
+4. You may have some little problem using update functions of controller-utils package (CreateOrUpdate(), Update(), Status().Update()).
+This is due to the fact that all those function control the version number of the resources to be updated. To overcome this problem it is possibile to use "Patch Version" of those: CreateOrPatch(), Patch(), Status().Patch() that do not make any check on the version.
```

Comment:
Is it good that the version is not checked? What issues could it lead to? Please add also an example of how a patch operation can be performed (it does not seem intuitive to me).